deepcell/applications/nuclear_segmentation.py

```diff
@@ -23,66 +23,76 @@
-        backbone=backbone,
-        use_imagenet=False,
-        panoptic=True,
-        num_semantic_heads=2,
-        num_semantic_classes=[4, 4],
-        input_shape=input_shape,
-        num_classes=1,
-        backbone_levels=backbone_levels,
-        pyramid_levels=pyramid_levels,
-        anchor_params=anchor_params,
-        norm_method='whole_image')
-
-    if use_pretrained_weights:
-        if backbone == 'resnet50':
-            # '/data/models/resnet50_panoptic_train-val_DVV_V4.h5'
+                'model-weights/general_nuclear_train_batch_size_82800_resnet50_'
+                '8_epochs_c4b2167eb754923856bc84fb29074413.h5')
+
+
+class NuclearSegmentation(Application):
+
+    def __init__(self,
+                 use_pretrained_weights=True,
+                 model_image_shape=(128, 128, 1)):
+
+        self.model = PanopticNet('resnet50',
+                                 input_shape=model_image_shape,
+                                 norm_method='whole_image',
+                                 num_semantic_heads=3,
+                                 num_semantic_classes=[1, 1, 2],
+                                 location=True,
+                                 include_top=True)
+
+        if use_pretrained_weights:
             weights_path = get_file(
-                'resnet50_panoptic_nuclear_segmentation.h5',
+                os.path.basename(WEIGHTS_PATH),
                 WEIGHTS_PATH,
                 cache_subdir='models',
-                md5_hash='6e925c49cb05a1e3b0e2210220922445')
+                md5_hash='eb29808ef2f662fb3bcda6986e47f91a'
+            )
 
-            model.load_weights(weights_path)
+            self.model.load_weights(weights_path)
         else:
-            raise ValueError('Backbone %s does not have a weights file.' %
-                             backbone)
+            weights_path = None
+
+        dataset_metadata = {
+            'name': 'general_nuclear_train_large',
+            'other': 'Pooled nuclear data from HEK293, HeLa-S3, NIH-3T3, and RAW264.7 cells.'
+        }
+
+        model_metadata = {
+            'batch_size': 16,
+            'lr': 1e-4,
+            'lr_decay': 0.95,
+            'training_seed': 0,
+            'n_epochs': 8,
+            'training_steps_per_epoch': 82800 // 16,
+            'validation_steps_per_epoch': 20760 // 16
+        }
+
+        super(NuclearSegmentation, self).__init__(self.model,
+                                                  model_image_shape=model_image_shape,
+                                                  model_mpp=0.65,
+                                                  preprocessing_fn=None,
+                                                  postprocessing_fn=deep_watershed,
+                                                  dataset_metadata=dataset_metadata,
+                                                  model_metadata=model_metadata)
 
-    return model
+        self.predict = self._predict_segmentation
```

Comment:
I think it would be clearer to redefine `predict` than to assign it in the init.